pkg/apis/hub/v1alpha1/api_portal_auth.go

```diff
@@ -95,6 +101,102 @@ type ClaimsSpec struct {
+type LDAPConnectionConfig struct {
+	// URL is the URL of the LDAP server, including the protocol (ldap or ldaps) and the port.
+	// +kubebuilder:validation:XValidation:message="must be a valid LDAP URL",rule="isURL(self) && (self.startsWith('ldap://') || self.startsWith('ldaps://'))"
+	URL string `json:"url"`
+
+	// StartTLS instructs the middleware to issue a StartTLS request when initializing the connection with the LDAP server.
+	// +optional
+	StartTLS bool `json:"startTls,omitempty"`
+
+	// InsecureSkipVerify controls whether the server's certificate chain and host name is verified.
+	// +optional
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	// CertificateAuthority is a PEM-encoded certificate to use to establish a connection with the LDAP server if the
+	// connection uses TLS but that the certificate was signed by a custom Certificate Authority.
+	// +optional
+	CertificateAuthority string `json:"certificateAuthority,omitempty"`
+
+	// BindDN is the domain name to bind to in order to authenticate to the LDAP server when running in search mode.
+	// If empty, an anonymous bind will be done.
+	// +optional
+	BindDN string `json:"bindDn,omitempty"`
+
+	// BindPasswordSecretName is the name of the Kubernetes Secret containing the password for the bind DN.
+	// The secret must contain a key named 'password'.
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	BindPasswordSecretName string `json:"bindPasswordSecretName,omitempty"`
+
+	// BaseDN is the base domain name that should be used for bind and search queries.
+	BaseDN string `json:"baseDn"`
+
+	// Attribute is the LDAP object attribute used to form a bind DN when sending bind queries.
+	// The bind DN is formed as <Attribute>=<Username>,<BaseDN>.
+	// +optional
+	// +kubebuilder:default="cn"
+	Attribute string `json:"attribute,omitempty"`
+
+	// SearchFilter is used to filter LDAP search queries.
+	// Example: (&(objectClass=inetOrgPerson)(gidNumber=500)(uid=%s))
+	// %s can be used as a placeholder for the username.
+	// +optional
+	SearchFilter string `json:"searchFilter,omitempty"`
+}
+
+// PortalAuthLDAPConfig holds LDAP configuration for portal authentication.
+type PortalAuthLDAPConfig struct {
+	LDAPConnectionConfig `json:",inline"`
+
+	// Groups configures group extraction.
+	// +optional
+	Groups *LDAPGroups `json:"groups,omitempty"`
+
+	// Attributes configures LDAP attribute mappings for user attributes.
+	// +optional
+	Attributes *LDAPAttributesSpec `json:"attributes"`
+
+	// SyncedAttributes is a list of additional attributes to sync from the OIDC provider.
+	// Each attribute must correspond to a configured claim field.
+	// +optional
+	// +kubebuilder:validation:MaxItems=6
+	// +kubebuilder:validation:items:Enum=groups;userId;firstname;lastname;email;company
+	SyncedAttributes []string `json:"syncedAttributes,omitempty"`
+}
+
+// LDAPAttributesSpec configures LDAP attribute mappings for user attributes.
+type LDAPAttributesSpec struct {
```

Comment:
Is the `Spec` suffix really needed? WDYT about renaming it to `LDAPAttributes`?

Reply:
No it's not, I simply follow the naming pattern used in OIDC: `ClaimSpec`. But I agree its not great. Let me fix this quickly